[quant][fx] Update name of packed weight attributes

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#51259 [quant][fx] Update name of packed weight attributes**

Summary:

Store the FQN of the module that is using the packed weights (the quantized op)

In the case of fusion we update the scope mapping to store the module path of the fused node.

Test Plan:
python test/test_quantization.py test_packed_weight_fused_op

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26117964](https://our.internmc.facebook.com/intern/diff/D26117964)

supriyar committed:
[quant][fx] Update name of packed weight attributes
Summary:

Store the FQN of the module that is using the packed weights (the quantized op)

In the case of fusion we update the scope mapping to store the module path of the fused node.

Test Plan:
python test/test_quantization.py test_packed_weight_fused_op

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant][fx] Update name of packed weight attributes"


Summary:

Store the FQN of the module that is using the packed weights (the quantized op)

In the case of fusion we update the scope mapping to store the module path of the fused node.

Test Plan:
python test/test_quantization.py test_packed_weight_fused_op

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26117964](https://our.internmc.facebook.com/intern/diff/D26117964)

[ghstack-poisoned]

jerryzh168 commented on torch/quantization/fx/quantization_patterns.py:
```diff
@@ -327,8 +327,10 @@ def convert(self, quantizer: QuantizerCls, node: Node, load_arg: Callable,
-                    return quantizer.quantized_graph.create_node(
+                    op = quantizer.quantized_graph.create_node(
                         'call_function', qconv_op, qconv_args, kwargs)
+                    quantizer.node_name_to_scope[op.name] = quantizer.node_name_to_scope[self.conv_node.name]
```
can you add a comment here?

jerryzh168 replied:
Actually can you add a TODO here:

"TODO: may need to change the key to Node regenerate the map in each transformationo since we might not be able to rely on the name"

supriyar committed:
Update on "[quant][fx] Update name of packed weight attributes"


Summary:

Store the FQN of the module that is using the packed weights (the quantized op)

In the case of fusion we update the scope mapping to store the module path of the fused node.

Test Plan:
python test/test_quantization.py test_packed_weight_fused_op

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26117964](https://our.internmc.facebook.com/intern/diff/D26117964)

[ghstack-poisoned]

supriyar committed:
Update on "[quant][fx] Update name of packed weight attributes"


Summary:

Store the FQN of the module that is using the packed weights (the quantized op)

In the case of fusion we update the scope mapping to store the module path of the fused node.

Test Plan:
python test/test_quantization.py test_packed_weight_fused_op

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26117964](https://our.internmc.facebook.com/intern/diff/D26117964)

[ghstack-poisoned]
